### PR TITLE
Clarify usage of patternfile roots

### PR DESCRIPTION
**Pull request by bobthebadguy**
See #6231. Clarifies what patternfile roots actually do with examples.

### CODE REVIEW

File: docs/usage/help.rst.inc [hunk 1]
```diff
@@ -200,6 +200,18 @@ Examples::
...
+        borg create --exclude /home/bobby/junk repo::arch /home/bobby /home/susan
+
+    Or you could do that with a patternfile like so, replacing roots specified on the command line with ones in the file::
+
+        R /home/bobby
+        R /home/susan
+
+        # Note that excludes use fm: by default and patternfiles use sh: by default.
+        - /home/bobby/junk
+
```
Comment (ThomasWaldmann) [inline docs/usage/help.rst.inc:214]: maybe add the borg command that uses this pattern file.

**Comment by bobthebadguy** [comment]
I still feel like this could be a bit clearer

File: docs/usage/help.rst.inc [hunk 2]
```diff
@@ -149,7 +149,7 @@ Examples::
 
     A more general and easier to use way to define filename matching patterns exists
     with the experimental ``--pattern`` and ``--patterns-from`` options. Using these, you
-    may specify the backup roots (starting points) and patterns for inclusion/exclusion.
+    may specify the backup roots (starting points), pattern files, and patterns for inclusion/exclusion.
```
Comment (ThomasWaldmann) [inline docs/usage/help.rst.inc:152]: this paragraph is talking about the patterns specifically (not the command line in general), so we are already "in" the pattern file, so you can't specify a pattern file there.

so I'ld suggest to just remove this change.

File: docs/usage/help.rst.inc [hunk 3]
```diff
@@ -195,11 +195,30 @@ Examples::
...
+        borg create --patterns-from patternfile.lst repo::arch
+
+    The patternfile::
+
+        R /home/bobby
+        R /home/susan
+
+        # note that excludes use fm: by default and patternfiles use sh: by default.
+        - /home/bobby/junk
+
```
Comment (ThomasWaldmann) [inline docs/usage/help.rst.inc:219]: you say "same thing", so people might expect this is precisely doing the same thing as the previous example, which it does not.

maybe adjust it so it really does exactly the same thing?

**Review by ThomasWaldmann — CHANGES_REQUESTED** [review]
most important: read the comment at the top of the file you are editing.

just use full-text search the right place, likely it is in `src/borg/archiver.py`.

**Comment by bobthebadguy** [comment]
Did I mess up the indentation? 
Sorry about that, should have read more closely

File: src/borg/archiver.py [hunk 4]
```diff
@@ -2412,10 +2412,31 @@ def do_break_lock(self, args, repository):
...
                 + /home/susan
+                # also back up this exact file
+                + pf:/home/bobby/specialfile.txt
                 # don't backup the other home directories
                 - /home/*
                 # don't even look in /proc
-                ! /proc\n\n''')
+                ! /proc
+
+            As another example, you could do the same thing in either of these ways::
```
Comment (ThomasWaldmann) [inline src/borg/archiver.py:2422]: "same" might get misunderstood as a backreference to the previous example (lines before 2420).

maybe rather say:

Here is another example that shows how recursion roots and excludes can be given either directly on the command line or in a patternfile.

(with that, you could drop the comments from 2424 and 2426)

File: src/borg/archiver.py [hunk 5]
```diff
@@ -2412,10 +2412,31 @@ def do_break_lock(self, args, repository):
...
+
+                # note that excludes use fm: by default and patternfiles use sh: by default.
+                # therefore, we need to specify fm: to have the same exact behavior.
+                P fm
+                R /home/bobby
+                R /home/susan
+
+                - /home/bobby/junk
+
+            This allows you to share the same patterns between multiple repositories without needing to specify them on the command line.\n\n''')
```
Comment (ThomasWaldmann) [inline src/borg/archiver.py:2439]: borg can output this help directly on the terminal.
if a line is longer than 80 chars, rather continue on next line.

**Comment by ThomasWaldmann** [comment]
there seems to be a conflict now with other master branch changes.

and because you have committed to directly to your master branch, this is a little tricky to fix now...

maybe just tell when you've finished your changes and i'll try to fix it.

**Comment by ThomasWaldmann** [comment]
now indentation is wrong again. most stuff will appear inside a code block now instead as normal text.

**Comment by ThomasWaldmann** [comment]
```
cd docs ; make clean ; make html
```
then look at the local html files with a browser.

**Comment by bobthebadguy** [comment]
It looks like that fixed it?

Thanks for all your help with this and sorry for the trouble

**Review by ThomasWaldmann — APPROVED** [review]
LGTM

**Comment by ThomasWaldmann** [comment]
Thanks!

**Comment by hexagonrecursion** [comment]
I plan to backport this today

**Pull request merged**